allensdk/brain_observatory/behavior/session_apis/data_transforms/behavior_ophys_data_transforms.py

```diff
@@ -138,12 +140,64 @@ def get_sync_data(self):
         sync_path = self.extractor.get_sync_file()
         return get_sync_data(sync_path)
 
-    @memoize
-    def get_stimulus_timestamps(self):
+    def _load_stimulus_timestamps_and_delay(self):
+        """
+        Load the stimulus timestamps (uncorrected for
+        monitor delay) and the monitor delay
+        """
         sync_path = self.extractor.get_sync_file()
-        timestamps, _, _ = (OphysTimeAligner(sync_file=sync_path)
-                            .corrected_stim_timestamps)
-        return np.array(timestamps)
+        aligner = OphysTimeAligner(sync_file=sync_path)
+        (self._stimulus_timestamps,
+         delta) = aligner.clipped_stim_timestamps
+
+        try:
+            delay = aligner.monitor_delay
+        except ValueError as ee:
+            rig_name = self.get_metadata()['rig_name']
+
+            err_msg = ee.args[0]
+            warning_msg = 'Monitory delay calculation failed '
+            warning_msg += 'with ValueError\n'
+            warning_msg += f'    "{ee}"'
+            warning_msg += '\nlooking monitor delay up from table '
+            warning_msg += f'for rig: {rig_name} '
+
+            # see
+            # https://github.com/AllenInstitute/AllenSDK/issues/1318
+            # https://github.com/AllenInstitute/AllenSDK/issues/1916
+            delay_lookup = {'CAM2P.1': 0.020842,
```

Comment:
Is there not a way to look this information up from LIMS? Having this table built into the function makes me uneasy...

Reply:
Based on the issue threads, I think Wayne read in a bunch of sessions from LIMS, calculated their monitor delays, and averaged over them.

Which makes me think the answer is "no"

Reply:
Hmm... can you create a ticket to revisit ways of getting these values from LIMS in the future? There will be a MESO.2 rig eventually I believe and we can't just keep adding to the table...

Reply:
Based on this comment

https://app.zenhub.com/workspaces/allensdk-10-5c17f74db59cfb36f158db8c/issues/alleninstitute/allensdk/1916#issuecomment-782297075

it looks like the only LIMS query we can run would be an average, which will change over time as more runs get ingested. It won't change by much (we hope), but it will change. I'm not sure that's ideal, either.

Reply:
That's what the behavior_ophys_write_nwb_strategy.rb from LIMS is for. It will generate a JSON where this value will be frozen that is used as input to these DataTransformers classes.

Reply:
But like I said, this stuff can be the scope for a different ticket. I'm not asking for any changes here.

Reply:
(I think our posts crossed in the ether; I'll create a ticket)

Reply:
https://app.zenhub.com/workspaces/allensdk-10-5c17f74db59cfb36f158db8c/issues/alleninstitute/allensdk/1928